json_test.go

```diff
@@ -18,62 +19,82 @@ func TestMarshalJSON(t *testing.T) {
 	// . none of Google's WKT messages fit the bill :(
 	t.Run("default", func(t *testing.T) {
 		ts := timestamppb.Now()
-		expected := fmt.Sprintf(`{"seconds":"%d","nanos":%d}`, ts.GetSeconds(), ts.GetNanos())
+		// https://github.com/protocolbuffers/protobuf-go/blob/32051b4f86e54c2142c7c05362c6e96ae3454a1c/encoding/protojson/well_known_types.go#L764-L775
+		expected := fmt.Sprintf("\"%s\"", ts.AsTime().Format("2006-01-02T15:04:05.000000Z"))
 
 		res, err := csproto.JSONMarshaler(ts).MarshalJSON()
 
 		assert.NoError(t, err)
 		assert.JSONEq(t, expected, string(res))
 	})
 	t.Run("with-indent", func(t *testing.T) {
-		ts := timestamppb.Now()
-		expected := fmt.Sprintf("{\n  \"seconds\":\"%d\",\n  \"nanos\":%d\n}", ts.GetSeconds(), ts.GetNanos())
+		input, err := structpb.NewValue(map[string]interface{}{
+			"firstName": "John",
+			"lastName":  "Smith",
+		})
+		assert.NoError(t, err)
+
+		expected := "{\n  \"firstName\":\"John\",\n  \"lastName\":\"Smith\"\n}"
 
 		opts := []csproto.JSONOption{
 			csproto.JSONIndent("  "),
 		}
-		res, err := csproto.JSONMarshaler(ts, opts...).MarshalJSON()
+		res, err := csproto.JSONMarshaler(input, opts...).MarshalJSON()
 
 		assert.NoError(t, err)
 		assert.JSONEq(t, expected, string(res))
 	})
 	t.Run("exclude-zero-values", func(t *testing.T) {
-		ts := timestamppb.Now()
-		ts.Nanos = 0
-		expected := fmt.Sprintf(`{"seconds":"%d"}`, ts.GetSeconds())
+		input, err := structpb.NewValue(map[string]interface{}{
+			"firstName": "John",
+			"lastName":  "Smith",
+			"location":  nil,
```

Comment:
@dylan-bourque Both zero-value tests are currently failing and I'm not sure how to write a good test case for this 🤔 -- any ideas? 

In both instances, the zero-value `nil` is being marshaled into the JSON output.

Reply:
This is because the behavior of a `structpb.Struct` is different than `timestamppb.Timestamp`.  It's marshaled as a map with no check for empty values ([link](https://github.com/protocolbuffers/protobuf-go/blob/32051b4f86e54c2142c7c05362c6e96ae3454a1c/encoding/protojson/encode.go#L329-L344)).

We'll need to find a different "message" type for these tests.  Looking at the code in `well_known_types.go` I don't see any logic for excluding zero values for any of the WKTs.  I was really trying to avoid having a `.proto` file here and generating code just for these tests but that may be the only option. 😢

Reply:
That makes sense. It sounds like our only option to improve test coverage in JSON marshaling will be to add a test proto. I'll go ahead and do that, and additional fix the TODO about testing JSONUseEnumNumbers.